raphtory/src/db/api/view/serialise.rs

```diff
@@ -0,0 +1,1092 @@
+        prop::Value::F64(f) => Prop::F64(*f),
+        prop::Value::Str(s) => Prop::Str(intern_str(intern_map, s)),
+        prop::Value::Prop(props) => Prop::List(Arc::new(
+            props
+                .properties
+                .iter()
+                .map(|prop| as_prop_value(prop.value.as_ref(), intern_map))
+                .collect(),
+        )),
+        prop::Value::Map(dict) => Prop::Map(Arc::new(
+            dict.map
+                .iter()
+                .map(|(k, v)| {
+                    (
+                        intern_str(intern_map, k),
+                        as_prop_value(v.value.as_ref(), intern_map),
+                    )
+                })
+                .collect(),
+        )),
+        serialise::prop::Value::NdTime(ndt) => {
+            let NdTime {
+                year,
+                month,
+                day,
+                hour,
+                minute,
+                second,
+                nanos,
+            } = ndt;
+            let ndt = NaiveDateTime::new(
+                NaiveDate::from_ymd_opt(*year as i32, *month as u32, *day as u32).unwrap(),
+                NaiveTime::from_hms_nano_opt(
+                    *hour as u32,
+                    *minute as u32,
+                    *second as u32,
+                    *nanos as u32,
+                )
+                .unwrap(),
+            );
+            Prop::NDTime(ndt)
+        }
+        serialise::prop::Value::DTime(dt) => {
+            Prop::DTime(DateTime::parse_from_rfc3339(dt).unwrap().into())
+        }
+        serialise::prop::Value::Graph(graph) => {
+            let g = Graph::new();
+            Graph::decode_from_bytes(&graph, &g).expect("Failed to decode graph");
+            Prop::Graph(g)
+        }
+        serialise::prop::Value::PersistentGraph(graph) => {
+            let g = Graph::new().persistent_graph();
+            PersistentGraph::decode_from_bytes(&graph, &g).expect("Failed to decode graph");
+            Prop::PersistentGraph(g)
+        }
+        serialise::prop::Value::DocumentInput(doc) => Prop::Document(DocumentInput {
+            content: doc.content.clone(),
+            life: doc
+                .life
+                .as_ref()
+                .map(|l| match l.l_type {
+                    Some(lifespan::LType::Interval(lifespan::Interval { start, end })) => {
+                        Lifespan::Interval { start, end }
+                    }
+                    Some(lifespan::LType::Event(lifespan::Event { time })) => {
+                        Lifespan::Event { time }
+                    }
+                    None => Lifespan::Inherited,
+                })
+                .unwrap_or(Lifespan::Inherited),
+        }),
+    };
+    value
+}
+
+fn intern_str(intern_map: &mut HashMap<String, ArcStr>, s: &String) -> ArcStr {
+    let s = intern_map.entry(s.clone()).or_insert_with(|| {
+        let s: ArcStr = s.clone().into();
+        s
+    });
+    s.clone()
+}
```

Comment:
why did you rebuild the `String` interning? You can just use `graph.process_prop_value(prop)` for this?